### DISCUSSION TITLE
Runtime Configuration + getStaticProps

### DISCUSSION THREAD

**Discussion by iondrimba** [vercel/next.js#25473] [Help] [answered]
I'm currently upgrading an application to use **next-i18next@8.3.0** and according the docs I should use this implementation:

```
import { serverSideTranslations } from 'next-i18next/serverSideTranslations'

export const getStaticProps = async ({ locale }) => ({
  props: {
    ...await serverSideTranslations(locale, ['common', 'footer']),
  }
})
```
My question is,  can ` Runtime Configuration `  work with this implementation of `getStaticProps`? 
According to the Nextjs doc, ` Runtime Configuration ` only works with `getInitialProps`

![image](https://user-images.githubusercontent.com/178548/119578786-8d18ff80-bd93-11eb-915a-7204ed4abe89.png)

### Current implementation
```
import getConfig from 'next/config'

const { publicRuntimeConfig } = getConfig();

const Page = () => {
  return (
    <h1>Page</h1>
  );
};

Page.getInitialProps = async () => {};

export default Page;

```

### New implementation
```
import getConfig from 'next/config'

const { publicRuntimeConfig } = getConfig();

const Page = () => {
  return (
    <h1>Page</h1>
  );
};

export const getStaticProps = async () => {};

export default Page;
```

**Comment by ParallelUniv3rse** [comment, marked as answer]
No. GetStaticProps only runs when you generate your app via `next build`. GetStaticProps does not run when the server starts. As the docs say - it can be only accessed in getInitialProps (and possibly getServerSideProps according to https://github.com/vercel/next.js/discussions/25792) because only those functions can run when server is started (at runtime).

**Reply by iondrimba** [reply]
Thanks!
